storage/redis/redis.go

```diff
@@ -0,0 +1,545 @@
+		err = json.Unmarshal([]byte(val), &password)
+		if err != nil {
+			return passwords, err
+		}
+		passwords = append(passwords, password)
+	}
+	return passwords, nil
+}
+
+func (c *client) CreateOfflineSessions(s storage.OfflineSessions) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	return c.createKey(ctx, keySession(s.UserID, s.ConnID), s)
+}
+
+func (c *client) UpdateOfflineSessions(userID string, connID string, updater func(s storage.OfflineSessions) (storage.OfflineSessions, error)) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	var val storage.OfflineSessions
+	return c.updateKey(ctx, keySession(userID, connID), &val, func(old interface{}) (interface{}, error) {
+		newVal, err := updater(*old.(*storage.OfflineSessions))
+		return newVal, err
+	})
+}
+
+func (c *client) GetOfflineSessions(userID string, connID string) (s storage.OfflineSessions, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	if err = c.getKey(ctx, keySession(userID, connID), &s); err != nil {
+		return
+	}
+	return s, nil
+}
+
+func (c *client) DeleteOfflineSessions(userID string, connID string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	return c.deleteKey(ctx, keySession(userID, connID))
+}
+
+func (c *client) CreateConnector(connector storage.Connector) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	return c.createKey(ctx, keyID(connectorPrefix, connector.ID), connector)
+}
+
+func (c *client) GetConnector(id string) (client storage.Connector, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	err = c.getKey(ctx, keyID(connectorPrefix, id), &client)
+	return client, err
+}
+
+func (c *client) UpdateConnector(id string, updater func(s storage.Connector) (storage.Connector, error)) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	var val storage.Connector
+	return c.updateKey(ctx, keyID(connectorPrefix, id), &val, func(old interface{}) (interface{}, error) {
+		newVal, err := updater(*old.(*storage.Connector))
+		return newVal, err
+	})
+}
+
+func (c *client) DeleteConnector(id string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+	return c.deleteKey(ctx, keyID(connectorPrefix, id))
+}
+
+func (c *client) ListConnectors() (connectors []storage.Connector, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultStorageTimeout)
+	defer cancel()
+
+	kvs, err := c.getKvs(ctx, connectorPrefix)
+	if err != nil {
+		return nil, err
+	}
+	for _, val := range kvs {
+		var connector storage.Connector
+		err = json.Unmarshal([]byte(val), &connector)
```

Comment:
storage.Connector has config json field of type []byte. If we have config json object stored in the redis, then json unmarshaling returns an error.

"{\"type\":\"ldap\",\"name\":\"OpenLDAP\",\"id\":\"ldap\",\"config\":{\"host\":\"localhost:389\",\"insecureNoSSL\":true,\"bindDN\":\"cn=admin,dc=example,dc=org\",\"bindPW\":\"admin\",\"usernamePrompt\":\"EmailAddress\",\"userSearch\":{\"baseDN\":\"ou=People,dc=example,dc=org\",\"filter\":\"(objectClass=person)\",\"username\":\"mail\",\"idAttr\":\"DN\",\"emailAttr\":\"mail\",\"nameAttr\":\"cn\"},\"groupSearch\":{\"baseDN\":\"ou=Groups,dc=example,dc=org\",\"filter\":\"(objectClass=groupOfNames)\",\"userMatchers\":[{\"userAttr\":\"DN\",\"groupAttr\":\"member\"}],\"nameAttr\":\"cn\"}}}"

Issue is fixed after declaring config of type json.RawMessage instead of []byte.